### DISCUSSION TITLE
文本检测漏行问题

### DISCUSSION THREAD

**Discussion by dragronstranding** [PaddlePaddle/PaddleOCR#14460] [Q&A]
### 🔎 Search before asking

- [X] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [X] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [X] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

![image](https://github.com/user-attachments/assets/67d0d01c-7dd2-4df3-8575-2e904db33dab)
例如上图，不管是用v3还是v4的文本检测里会漏掉最上面的“APP 金东纸业。。。”那一行

### 🏃‍♂️ Environment (运行环境)

paddleocr=2.9.2
paddlepaddle=3.0.0b1

### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

# 初始化 PaddleOCR
paddleocr = PaddleOCR(lang='ch', use_angle_cls=False, show_log=True,
                      det_model_dir='../weights/ch_PP-OCRv4_det_infer',
                      rec_model_dir='../weights/ch_PP-OCRv4_rec_infer'
                      )

**Comment by jingsongliujing** [comment]
看上去是字体有部分歪曲和模糊，可以通过预处理（比如二值化，图像缩放和归一化、图像增强）或者标注数据再训练的方式提高识别精度

**Comment by GreatV** [comment]
根据问题描述，您遇到了在使用 PaddleOCR 进行文本检测时漏掉某些行（例如顶部的“APP 金东纸业……”）的问题。这种情况在复杂布局或者文本区域较小的图片中较为常见。以下是一些可能的解决方案和优化建议：

---

### 1. **调整检测模型参数**
PaddleOCR 提供了多个参数来调整检测效果，您可以尝试以下方法：
- **增大 `det_db_box_thresh` 和 `det_db_thresh` 的值**：这些参数控制文本框的检测阈值，适当调高可能会捕捉到更多文本区域。
  ```python
  paddleocr = PaddleOCR(
      lang='ch',
      det_model_dir='../weights/ch_PP-OCRv4_det_infer',
      rec_model_dir='../weights/ch_PP-OCRv4_rec_infer',
      det_db_box_thresh=0.4,  # 默认 0.6，尝试降低至 0.4
      det_db_thresh=0.2       # 默认 0.3，尝试降低至 0.2
  )
  ```
- **调整 `det_db_unclip_ratio`**：该参数控制检测框的扩展比例，适当增大可能会捕捉到更大的文本区域。
  ```python
  det_db_unclip_ratio=2.0  # 尝试从默认的 1.5 提高到 2.0
  ```

---

### 2. **使用更强大的检测算法**
PaddleOCR 支持多种检测算法，例如 `DB` 和 `SAST`。如果 `PP-OCRv4` 的检测效果不理想，可以尝试以下方法：
- **切换到 `SAST` 算法**，适用于复杂布局的场景。
  ```python
  paddleocr = PaddleOCR(
      lang='ch',
      det_algorithm='SAST',                 # 使用 SAST 检测算法
      det_model_dir='../weights/sast_infer' # SAST 模型路径
  )
  ```

- **升级 PaddleOCR 模型**：确保使用的是最新版本的模型（例如 v4），并尝试其他模型如 `PP-OCRv3`，以便比较检测效果。

---

### 3. **图片预处理**
图片质量对 OCR 的检测效果有很大影响，尝试以下操作：
- **裁剪图片**：将目标文本行放大到图片的中央，提供更清晰的输入。
- **增强对比度**：使用 OpenCV 或 PIL 增强图片对比度，减少噪声干扰。
  ```python
  from PIL import Image, ImageEnhance

  img = Image.open('image.jpg').convert('RGB')
  enhancer = ImageEnhance.Contrast(img)
  img_enhanced = enhancer.enhance(2)  # 增强对比度
  img_enhanced.save('enhanced_image.jpg')
  ```

---

### 4. **启用角度分类器**
角度分类器可以帮助更好地识别倾斜文本。如果未启用，请确保设置 `use_angle_cls=True`：
```python
paddleocr = PaddleOCR(
    lang='ch',
    use_angle_cls=True,  # 启用角度分类器
    det_model_dir='../weights/ch_PP-OCRv4_det_infer',
    rec_model_dir='../weights/ch_PP-OCRv4_rec_infer'
)
```

---

### 5. **检查图片分辨率**
PaddleOCR 默认的 `det_limit_side_len` 为 960。如果图片分辨率较低，某些细节可能会丢失。尝试增大该值：
```python
paddleocr = PaddleOCR(
    lang='ch',
    det_limit_side_len=1920  # 增大图片限制长度
)
```

---

### 6. **参考社区讨论**
在 [PaddleOCR Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions/13849) 中，有用户反馈了类似的“文本漏检”问题，建议您参考以下解决方案：
- **训练自定义检测模型**：如果以上方法仍然无法解决问题，可以基于您自己的数据集进行模型微调，从而提高模型在特定场景下的效果。
  官方教程：[PaddleOCR 自定义训练文档](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.9/doc/doc_ch/multi_language.md)。

---

### 总结
根据您的描述，漏检可能是由于默认参数设置不适合您的具体场景。建议依次尝试以下方法：
1. 调整检测参数（如 `det_db_box_thresh`、`det_db_unclip_ratio`）。
2. 切换至更强的检测算法（如 `SAST`）。
3. 对图片进行预处理（如增强对比度、调整分辨率）。
4. 如果问题依然存在，考虑训练自定义模型以适配场景。

如果仍有问题，欢迎在 [PaddleOCR Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) 提交详细的反馈。

---
*Response generated by [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*
